### PR TITLE
[VL] Enable Spark3.4 linear-regression.sql test case in GlutenSQLQueryTestSuite

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

Enable `linear-regression.sql` in GlutenSQLQueryTestSuite

## How was this patch tested?

`linear-regression.sql` in`GlutenSQLQueryTestSuite`.

### CODE REVIEW

**Comment by liujiayi771** [comment]
cc @PHILO-HE @rui-mo, thanks.

File: gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxSQLQueryTestSettings.scala [hunk 1]
```diff
@@ -240,6 +240,8 @@ object VeloxSQLQueryTestSettings extends SQLQueryTestSettings {
     // -- Aggregate with nulls.
     "group-by.sql",
     "udf/udf-group-by.sql",
+    // Overwrite some results of regr_intercept, regr_r2, corr.
+    "linear-regression.sql",
```
Comment (Yohahaha) [inline gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxSQLQueryTestSettings.scala:244]: the result of these functions is different with vanilla spark?

Comment (liujiayi771) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxSQLQueryTestSettings.scala:244]: Yes, most statistical functions have differences in precision. We have already fixed these results before, like https://github.com/apache/incubator-gluten/pull/5151.

**Review by Yohahaha — APPROVED** [review]
thank you!

**Pull request merged**